erroranalysis/erroranalysis/_internal/error_analyzer.py

```diff
@@ -130,11 +146,28 @@ def get_diff(self):
 
 
 class ModelAnalyzer(BaseAnalyzer):
-    def __init__(self, model, dataset, true_y, feature_names,
-                 categorical_features):
+    def __init__(self,
+                 model,
+                 dataset,
+                 true_y,
+                 feature_names,
+                 categorical_features,
+                 model_task=ModelTask.UNKNOWN,
+                 metric=None):
         self._model = model
-        super(ModelAnalyzer, self).__init__(dataset, true_y, feature_names,
-                                            categorical_features)
+        if model_task == ModelTask.UNKNOWN:
+            # Try to automatically infer the model task
+            predict_proba_flag = hasattr(model, 'predict_proba')
```

Comment:
Don't most classification models have `predict_proba`? Is this preferable to requiring it as an arg from the user? Since it's optional it wouldn't be obvious that you have to provide it as a user who wants to do classification. Anyway, that's my 2ct 😄

Reply:
so we automatically can tell the task based on predict_proba existence but user can override that (for example SVC in scikit-learn doesn't).  Eventually we should use same model wrapper as in interpret-community but that was out of the scope of this PR:
https://github.com/interpretml/interpret-community/blob/master/python/interpret_community/common/model_wrapper.py